src/server/serverhandler.cpp

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "serverhandler.h"
+#include "serverconnection.h"
+#include "leakdetector.h"
+#include "logger.h"
+
+#include <QHostAddress>
+#include <QTcpSocket>
+
+namespace {
+Logger logger(LOG_SERVER, "ServerHandler");
+}
+
+constexpr int SERVER_PORT = 8754;
+
+ServerHandler::ServerHandler() {
+  MVPN_COUNT_CTOR(ServerHandler);
+
+  logger.log() << "Creating the server";
+
+  if (!listen(QHostAddress::Any, SERVER_PORT)) {
```

Comment:
Can we use here QHostAddress::LocalHost? 
I know we drop connections that are not localhost, but i can still syn-scan this.